src/components/steps/StepSearchParty.tsx

```diff
@@ -88,6 +90,16 @@ export function StepSearchParty({
   );
   const [ecData, setEcData] = useState<PartyData | null>(null);
 
+  useEffect(() => {
```

Comment:
this code is duplicated, move this logic in the father component, and then, in the two components, work with filterCategories state